### PR TITLE
add log transformation for scatterplot

### LINKED ISSUE #437: [User request] log scale on `tm_g_scatterplot`
FYI @kumamiao 

Please can we have an option on `tm_g_scatterplot` to set one or both of the axes to be displayed in log-scale.

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
Closes #437 

FYI @donyunardi

Test with roxygen examples

image
image
image

And reflected in the SRC:
image

And if you try to do something not very sensible...
image

Will need to also be merged into the new quosure PRs...

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ### Enhancements
 * Added the `teal.reporter` functionality to all modules.
+* Added log scale options in `tm_g_scatterplot`.
```
Comment (nikolas-burkoff) [inline NEWS.md:5]:
```suggestion
* Added log transformation options to `tm_g_scatterplot`.
```

File: R/tm_g_scatterplot.R [hunk 2]
```diff
@@ -236,12 +236,14 @@ ui_g_scatterplot <- function(id, ...) {
           data_extract_spec = args$x,
           is_single_dataset = is_single_dataset_value
         ),
+        checkboxInput(ns("log_x"), "Use log transformation for X variable", value = FALSE),
```
Comment (Polkas) [inline R/tm_g_scatterplot.R:239]:
```suggestion
        checkboxInput(ns("log_x"), "Use log transformation", value = FALSE),
```

File: R/tm_g_scatterplot.R [hunk 3]
```diff
@@ -236,12 +236,14 @@ ui_g_scatterplot <- function(id, ...) {
...
         teal.transform::data_extract_ui(
           id = ns("y"),
           label = "Y variable",
           data_extract_spec = args$y,
           is_single_dataset = is_single_dataset_value
         ),
+        checkboxInput(ns("log_y"), "Use log transformation for Y variable", value = FALSE),
```
Comment (Polkas) [inline R/tm_g_scatterplot.R:246]:
```suggestion
        checkboxInput(ns("log_y"), "Use log transformation", value = FALSE),
```

**Review by Polkas — APPROVED** [review]
LGTM, most important is the proper validation (> 0 values for log)

the only question could be the log scale, as for some statisticians a base is log10 not loge (the default R log).
log10(1e7) # = 7
log(1e7) # 16.1181

**Comment by donyunardi** [comment]
Wow, thanks for picking this up so quickly @Polkas @nikolas-burkoff!

Hi @kumamiao,
Could you please advise on which log scale to use? Is it okay to use the default R log?
> the only question could be the log scale, as for some statisticians a base is log10 not loge (the default R log).
log10(1e7) # = 7
log(1e7) # 16.1181

**Comment by kumamiao** [comment]
agree, if possible, would be great to add the flexibility to choose amongst log, log2 and log10 in R

**Comment by nikolas-burkoff** [comment]
> agree, if possible, would be great to add the flexibility to choose amongst log, log2 and log10 in R

Done
image

**Review by Polkas — CHANGES_REQUESTED** [review]
We find out the brush stops to work.

File: R/tm_g_scatterplot.R [hunk 4]
```diff
@@ -761,9 +845,10 @@ srv_g_scatterplot <- function(id,
     )
 
     output$data_table <- DT::renderDataTable({
-      # if not dependent on plot_r() it tries to print a table before chunks are populated with the correct ANL
+      # if not dependent on plot_r() initially it tries to print a table
+      # before chunks are populated with the correct ANL
       # in plot_r which ends up in a bunch of warnings that ANL is not in chunks
-      plot_r()
+      isolate(plot_r())
       plot_brush <- brush$brush()
```
Comment (nikolas-burkoff) [inline R/tm_g_scatterplot.R:852]: 🤯

**Comment by nikolas-burkoff** [comment]
> We find out the brush stops to work.

Should be fixed now

File: R/tm_g_scatterplot.R [hunk 5]
```diff
@@ -236,12 +236,32 @@ ui_g_scatterplot <- function(id, ...) {
           data_extract_spec = args$x,
           is_single_dataset = is_single_dataset_value
         ),
+        checkboxInput(ns("log_x"), "Use log transformation", value = FALSE),
+        conditionalPanel(
+          condition = paste0("input['", ns("log_x"), "'] == true"),
+          radioButtons(
+            ns("log_x_base"),
+            label = "Base",
```
Comment (Polkas) [inline R/tm_g_scatterplot.R:244]:
```suggestion
            label = NULL,
```

File: R/tm_g_scatterplot.R [hunk 6]
```diff
@@ -236,12 +236,32 @@ ui_g_scatterplot <- function(id, ...) {
...
           label = "Y variable",
           data_extract_spec = args$y,
           is_single_dataset = is_single_dataset_value
         ),
+        checkboxInput(ns("log_y"), "Use log transformation", value = FALSE),
+        conditionalPanel(
+          condition = paste0("input['", ns("log_y"), "'] == true"),
+          radioButtons(
+            ns("log_y_base"),
+            label = "Base",
```
Comment (Polkas) [inline R/tm_g_scatterplot.R:260]:
```suggestion
            label = NULL,
```

Comment (Polkas) [reply R/tm_g_scatterplot.R:260]: The base already in the 2/3 of names

File: R/tm_g_scatterplot.R (hunk 5 above)
Comment (Polkas) [reply R/tm_g_scatterplot.R:244]: The base already in the 2/3 of names

**Pull request merged**